static/app/views/alerts/rules/metric/ruleConditionsForm.tsx

```diff
@@ -142,6 +143,16 @@ class RuleConditionsForm extends PureComponent<Props, State> {
 
   get timeWindowOptions() {
     let options: Record<string, string> = TIME_WINDOW_MAP;
+    const {alertType, timeWindow, onTimeWindowChange} = this.props;
+
+    if (alertType === 'custom_metrics') {
+      // Do not show ONE MINUTE interval as an option for custom_metrics alert
+      options = omit(options, TimeWindow.ONE_MINUTE.toString());
+
+      if (timeWindow === TimeWindow.ONE_MINUTE) {
+        onTimeWindowChange(TimeWindow.FIVE_MINUTES);
+      }
```

Comment:
Imho a getter should not trigger a state update.

Other side effects like this are handled here:
https://github.com/getsentry/sentry/blob/ef282f16269785e57ade409b3d8f1f91015920ae/static/app/views/alerts/rules/metric/ruleForm.tsx#L512